gninasrc/gninavis/cnn_visualization.cpp

```diff
@@ -520,9 +520,9 @@ std::string cnn_visualization::get_xyz(const std::string &line) {
   std::string z = line.substr(46, 8);
 
   //avoid negative zeros in string representation
-  if (x == "-0.000") x = "0.000";
```

Comment:
the trim_left below will remove the inserted space anyway. This is disturbingly nonrobust.

Reply:
i agree, i considered making it more robust but figured since you let the original version slide i'd be lazy